### DISCUSSION TITLE
Release 4.1

### DISCUSSION THREAD

**Discussion by revoxhere** [revoxhere/duino-coin#1798] [General]
## What's new, what's changed...
* Added Traditional Chinese by @BX2AFO in https://github.com/revoxhere/duino-coin/pull/1739
* Added LAN functionality to the ESP Code by @dl1lmh in https://github.com/revoxhere/duino-coin/pull/1735
* Added serial disable option to the ESP Code by @virophagesp in https://github.com/revoxhere/duino-coin/pull/1763
* Added Traditional Chinese language switch by @BX2AFO in https://github.com/revoxhere/duino-coin/pull/1751
* Fixed divide by zero on hashrate calculation by @GerardoNevarez in https://github.com/revoxhere/duino-coin/pull/1767
* Fixed node connection timeout in ESP Code by @revoxhere in https://github.com/revoxhere/duino-coin/pull/1749
* Fixed ESP issues after SERIAL_PRINTING merge by @GerardoNevarez in https://github.com/revoxhere/duino-coin/pull/1769
* Fixed compilation error with 3.x.x espressif library version by @MatusOllah in https://github.com/revoxhere/duino-coin/pull/1797
* Updated README.md by @ZL1LAC in https://github.com/revoxhere/duino-coin/pull/1778
* Updated German translations by @EinWildesPanda in https://github.com/revoxhere/duino-coin/pull/1773
* Updated Slovak translations by @MatusOllah in https://github.com/revoxhere/duino-coin/pull/1796
* Updated AVR Miner Chech translations by @fpetru104 in https://github.com/revoxhere/duino-coin/pull/1729
* Updated PC Miner Chech translations @fpetru104 in https://github.com/revoxhere/duino-coin/pull/1730
* Updated copyright years by @Tech1k in https://github.com/revoxhere/duino-coin/pull/1736

## New Contributors - welcome and thanks for your input!
* @dl1lmh made their first contribution in https://github.com/revoxhere/duino-coin/pull/1735
* @virophagesp made their first contribution in https://github.com/revoxhere/duino-coin/pull/1763
* @MatusOllah made their first contribution in https://github.com/revoxhere/duino-coin/pull/1797

### ⚠️ NOTICE: .exe binaries may be detected as a virus by your antivirus software. This is a **false positive** caused by pyinstallers bootloader, read about it [here](https://stackoverflow.com/questions/43777106/program-made-with-pyinstaller-now-seen-as-a-trojan-horse-by-avg). Duino-Coin is not a virus.

**Full Changelog**: https://github.com/revoxhere/duino-coin/compare/4.0...4.1
As always, please report any encountered issues. Happy mining!

<hr /><em>This discussion was created from the release <a href='https://github.com/revoxhere/duino-coin/releases/tag/4.1'>Release 4.1</a>.</em>

**Comment by ratotimo** [comment]
Hello... 
I don't know if everyone is going through this... But my Esp32 don't stay for long... Everything initially works normally, but after about 15 minutes they disconnect. My Arduino Uno also has the same problem. Any tips to solve this problem?
